cmd/controller/certificates.go

```diff
@@ -232,6 +237,20 @@ func (c *Certificates) Init(ctx context.Context) error {
 	return eg.Wait()
 }
 
+func detectLocalIP() (string, error) {
+	addrs, err := net.LookupIP("localhost")
```

Comment:
I suspect that we need to do it twice, once for localhost, and once for the hostname itself? On my box, localhost resolves to 127.0.0.1, and my hostname resolves to 127.0.0.2